test/unittest/tests/block/test_block_api_write.cpp

```diff
@@ -0,0 +1,279 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_api]: block_write.c
+ * The block manager performs file operations on specific blocks which eventually gets translated
+ * onto files. This test suite aims to test the possible cases for the write(), read()
+ * and write_size() API operations.
+ */
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <filesystem>
+
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include "../wrappers/item_wrapper.h"
+
+const int DEFAULT_BLOCK_SIZE = 256;
```

Comment:
Similar to how Tod has implemented his tests I think we need a way of testing a range of different allocation sizes. What if we give it an alloc size of 1? Or 0? What if we give it a non multiple of 2 like 71? 

Also a best fit vs first fit should be tested. Though that would come in a different ticket, and be quite a complex test case.

Reply:
On the best fit vs first fit thing I think we need to discuss this topic on the "how to" guide. Effectively any config multiplies the number of test cases needing writing.

Reply:
The range of different allocation sizes is already implemented inside the test_block_file.cpp, let me know if that is sufficient testing coverage however. 

I have created [WT-13550](https://jira.mongodb.org/browse/WT-13550) for the best fit vs first fit test cases.